src/rdb_protocol/terms/sort.cc

```diff
@@ -83,9 +83,9 @@ class orderby_term_t : public op_term_t {
         /* Add a sorting to the table if we're doing indexed sorting. */
         } else if (index.has()) {
             rcheck(tbl_slice.has(), base_exc_t::LOGIC,
-                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE.");
+                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE, not on a `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`.  Make sure ORDER_BY comes before any transformations such as MAP or FILTER.");
```

Comment:
@danielmewes -- we usually use all-caps for type names in error messages.  We don't seem to write command names very often, but I thought we used the same convention for those.  (For example, `r.table('test').order_by(index: 'id').order_by(index: 'id')` produces `RethinkDB::RqlQueryLogicError: Cannot perform multiple indexed ORDER_BYs on the same table.`.)

Also, I'm not sure what the user-facing docs say, but `filter` is called a transformation internally, for the same reason `concat_map` is.